### PR TITLE
Added onRender and improve onError callbacks in Yandex and Adfox ads

### PR DESCRIPTION
**Pull request by Urkass**
- Added new callback `onRender`
- Changed `onError` logic

### CODE REVIEW

**Comment by Urkass** [comment]
I signed it!

**Comment by Urkass** [comment]
Done

**Comment by Urkass** [comment]
Hello, is anything wrong with this pr? What should I do to start review?

**Comment by Urkass** [comment]
ping

**Comment by Urkass** [comment]
Please, take a look at this PR.

**Comment by erwinmombay** [comment]
@zhouyx can you please take a look at this? It looks like we missed triaging this correctly

File: ads/yandex.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const renderTo = 'yandex_rtb';
  * @param {!Object} data
  */
 export function yandex(global, data) {
-  validateData(data, ['blockId'], ['data', 'isAdfox']);
+  validateData(data, ['blockId'], ['data', 'onRender', 'onError']);
```
Comment (zhouyx) [inline ads/yandex.js]: Question: To confirm does this mean you no longer support `data-is-adfox` param, but instead expect `data-on-render` and `data-on-error` param from config? If so, please also update the change in documentation. Thanks

File: ads/yandex.js [hunk 2]
```diff
@@ -49,13 +49,17 @@ function addToQueue(global, data) {
       data: data.data,
       async: true,
       onRender: () => {
-        // Move adfox queue
-        if (data.isAdfox && global.Ya.adfoxCode.onRender) {
-          global.Ya.adfoxCode.onRender();
+        if (data.onRender) {
```
Comment (zhouyx) [inline ads/yandex.js]: I am not clear how your `render()` function, work. So I want to confirm here what is the `data` object here? Just afraid that `data.onRender` and `data-onError` are both strings.

File: ads/yandex.js (hunk 1 above)
Comment (Urkass) [reply ads/yandex.js]: yeah, no longer support for `data-is-adfox` param and new callbacks params instead.
Updated docs.

File: ads/yandex.js (hunk 2 above)
Comment (Urkass) [reply ads/yandex.js]: `data` has `onRender` and `onError` callbacks.
There are used only for internal purposes in adfox module and are useless for users.
Added typeOfFunction check if user added them to config as strings. Thanks.

**Comment by zhouyx** [comment]
@Urkass Thanks for the Pull Request, and sorry for the delay there. This looks great. Please rebase to latest master to fix the percy test. Thanks.

**Comment by Urkass** [comment]
done

**Comment by zhouyx** [comment]
Thanks! PR merged.

**Pull request merged**